### PR TITLE
ec2 vpc peerings implemented as nodes instead of relations

### PR DESCRIPTION
**Pull request by krisek**
Hi, I think I did something wrong with the previous pull request, so here we go with a clean one. Apologies for the lost context now, but this way we can start (again) with a clean sheet.

### CODE REVIEW

**Review by achantavy — COMMENTED** [review]
Thanks for continuing the work on this, do you think it's time to deprecate https://github.com/lyft/cartography/blob/master/cartography/intel/aws/ec2/vpc_peering.py?

**Comment by krisek** [comment]
intel/aws/ec2/vpc_peering.py is definitely obsoleted by this PR, but if I remove it, Travis builds do not pass, hence I left the file in the repo

**Comment by krisek** [comment]
I discovered one foreign account property setting was missing.

File: cartography/intel/aws/ec2/vpc_peerings.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    MERGE (avpc:AWSVpc{id: vpc_peering.AccepterVpcInfo.VpcId})
+    ON CREATE SET avpc.firstseen = timestamp()
+    SET avpc.lastupdated = {aws_update_tag}, avpc.vpcid = vpc_peering.AccepterVpcInfo.VpcId
+
+    MERGE (rvpc:AWSVpc{id: vpc_peering.RequesterVpcInfo.VpcId})
+    ON CREATE SET rvpc.firstseen = timestamp()
+    SET rvpc.lastupdated = {aws_update_tag}, rvpc.vpcid = vpc_peering.RequesterVpcInfo.VpcId
+
+    MERGE (aaccount:AWSAccount{id: vpc_peering.AccepterVpcInfo.OwnerId})
+    ON CREATE SET aaccount.firstseen = timestamp(), aaccount.foreign = true
```
Comment (achantavy) [inline cartography/intel/aws/ec2/vpc_peerings.py:47]: I don't think this treatment of `foreign` is the same as the older vpc_peering.py. In that file, I believe `foreign` was used to denote AWS accounts that were not included in the set of accounts that we "initially knew about", if that makes sense. We can talk about this in chat too.

If you get stuck on what to do here, I think it's fine to not set `foreign` for now.

File: cartography/intel/aws/ec2/vpc_peerings.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    return client.describe_vpc_peering_connections()['VpcPeeringConnections']
+
+
+@timeit
+def load_vpc_peerings(neo4j_session, data, region, aws_account_id, aws_update_tag):
+
+    ingest_vpc_peerings = """
+    UNWIND {vpc_peerings} AS vpc_peering
+
+    MERGE (pcx:PeeringConnection:AWSPeeringConnection{id: vpc_peering.VpcPeeringConnectionId})
```
Comment (achantavy) [inline cartography/intel/aws/ec2/vpc_peerings.py:24]: Is it possible to attach `pcx` to `avpc` and then `avpc` to `aaccount` etc etc in this query? I think we should try to make sure AWS resource nodes are connected to their accounts as early as possible.

File: cartography/intel/aws/ec2/vpc_peerings.py [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+@timeit
+@aws_handle_regions
+def get_vpc_peerings(boto3_session, region):
+    client = boto3_session.client('ec2', region_name=region, config=get_botocore_config())
+    return client.describe_vpc_peering_connections()['VpcPeeringConnections']
+
+
+@timeit
+def load_vpc_peerings(neo4j_session, data, region, aws_account_id, aws_update_tag):
```
Comment (achantavy) [inline cartography/intel/aws/ec2/vpc_peerings.py:19]: Can you split this into one function per query to make it easier to test? Like one `load()` function that calls `load_{sub_object}()`?

**Review by achantavy — CHANGES_REQUESTED** [review]
I tested this manually end-to-end and it runs pretty fast :).

I left a comment on splitting the `load()` function to a few sub-parts, let me your thoughts.

File: cartography/intel/aws/ec2/vpc_peerings.py (hunk 2 above)
Comment (krisek) [reply cartography/intel/aws/ec2/vpc_peerings.py:24]: I hope I managed to fix this with the last commit #6dc5950

File: cartography/intel/aws/ec2/vpc_peerings.py (hunk 3 above)
Comment (krisek) [reply cartography/intel/aws/ec2/vpc_peerings.py:19]: I hope I managed to fix this with the last commit #6dc5950 I was not fully sure what you meant by split

**Comment by krisek** [comment]
This code got completely outdated.

**Pull request closed without merging**